src/libraries/Microsoft.Extensions.Logging.TraceSource/src/TraceSourceLoggerProvider.cs

```diff
@@ -53,10 +53,9 @@ public ILogger CreateLogger(string name)
             return new TraceSourceLogger(GetOrAddTraceSource(name));
         }
 
-        private DiagnosticsTraceSource GetOrAddTraceSource(string name)
-        {
-            return _sources.GetOrAdd(name, InitializeTraceSource);
-        }
+        private DiagnosticsTraceSource GetOrAddTraceSource(string name) =>
+            _sources.TryGetValue(name, out DiagnosticsTraceSource? source) ? source :
+            _sources.GetOrAdd(name, InitializeTraceSource(name));
```

Comment:
It feels like this is such a perf trap ... 99.9% of devs would write the code as it was originally.

Reply:
> The other option here is

This library has a netstandard2.0 target.